jax/experimental/compilation_cache/file_system_cache_test.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2021 Google LLC
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from absl.testing import absltest
+from file_system_cache import FileSystemCache
+import jax.test_util as jtu
+import tempfile
+
+class FileSystemCacheTest(jtu.JaxTestCase):
+
+  def test_new_directory_and_path(self):
+   with tempfile.TemporaryDirectory() as tmpdir:
+     file_system_cache_instance = FileSystemCache(tmpdir)
```

Comment:
Style thing: `file_system_cache_instance` is very descriptive, but it's too long in my opinion :) You can just use `cache`.